itou/gps/management/commands/import_advisor_information.py

```diff
@@ -0,0 +1,123 @@
+import logging
+from math import ceil
+
+from django.db import transaction
+
+from itou.gps.excel import parse_gps_import
+from itou.gps.models import FranceTravailContact
+from itou.users.enums import UserKind
+from itou.users.models import JobSeekerProfile, User
+from itou.utils.command import BaseCommand
+from itou.utils.iterators import chunks
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Import advisors from an Excel file (GPS)
+    """
+
+    help = "Import advisor contact information for existing job seeker profiles from GPS excel file."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--wet-run",
+            dest="wet_run",
+            action="store_true",
+            help="Persist the changes to contacts in the database.",
+        )
+
+    def handle(self, wet_run=False, **options):
+        # parse the excel import
+        df = parse_gps_import()[["nir", "prescriber_name", "prescriber_email"]]
+        logger.info(f"Imported {len(df)} rows from GPS export.")
+
+        # not all of the NIRs in the dataset are complete, so we treat them
+        df["nir"] = df["nir"].apply(lambda x: f"{x}{str(97 - int(x) % 97).zfill(2)}" if len(x) == 13 else x)
+
+        count_nir_invalid_length = (df["nir"].str.len() != 15).sum()
+        if count_nir_invalid_length:
+            logger.warning(
+                f"There are {count_nir_invalid_length} included NIR values of invalid length after treatment."
+            )
+
+        beneficiaries_pks = User.objects.filter(
+            kind=UserKind.JOB_SEEKER, jobseeker_profile__nir__in=df["nir"].tolist()
+        ).values_list("pk", flat=True)
+        logger.info(f"Matched {len(beneficiaries_pks)} users in the database")
+
+        chunks_total = ceil(len(beneficiaries_pks) * 0.001)
+        created_rows_count = 0
+        updated_rows_count = 0
+
+        # tracked for information : NIRs included in the dataset missing from the database
+        count_missing_users = 0
+
+        for chunk_idx, chunked_beneficiary_pks in enumerate(chunks(beneficiaries_pks, 1000), 1):
+            chunked_beneficiary_qs = User.objects.filter(id__in=chunked_beneficiary_pks)
+            count_missing_users += max(0, len(chunked_beneficiary_pks) - len(chunked_beneficiary_qs))
+
+            # NOTE: cannot use select_for_update below
+            # NotSupportedError: FOR UPDATE cannot be applied to the nullable side of an outer join
+            implicated_profiles = JobSeekerProfile.objects.filter(user_id__in=chunked_beneficiary_qs).select_related(
+                "advisor_information", "user"
```

Comment:
On n’utilise pas l’info `user`.
```suggestion
                "advisor_information"
```